### PR TITLE
Config enhancements

### PR DESCRIPTION
**Pull request by MilesBHuff**
I went through and enhanced various configs.  The commit messages should do a (hopefully) satisfactory job of explaining each change;  but please feel free to ask for additional clarification, suggest changes, or even reject some of the changes -- this is your project, after-all.

The ridiculously high number of lines changed is due to `package-lock.json` and `npm-shrinkwrap.json`.  I tried to make `git` count them as binaries so that I wouldn't inflate my line count;  but alas, GitHib does not seem to care.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,83 @@
+#NOTE: Run `git add : --renormalize` after editing this file.
```
Comment (jjppof) [inline .gitattributes:1]: Maybe we could remove the ones that will never appear in this project like sass, PHP, sql etc, for simplicity.

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,26 @@
-node_modules
-dist
-debug.log
```
Comment (jjppof) [inline .gitignore:3]: Please add this debug.log too, sometimes it magically appears...

Comment (MilesBHuff) [reply .gitignore:3]: Got it!

File: .gitattributes (hunk 1 above)
Comment (MilesBHuff) [reply .gitattributes:1]: Yeah, I had the same thought.  Ended-up leaving them all in there;  but I can take out the ones that aren't currently relevant to the project.

File: tsconfig.json [hunk 3]
```diff
@@ -1,23 +1,72 @@
...
+
+        // Library versions
+        "moduleResolution": "node",
         "module": "commonjs",
-        "noImplicitAny": false,
-        "target": "es2017",
+        "target": "ES6",
+
+        // Types
+        "lib": [
```
Comment (jjppof) [inline tsconfig.json:14]: These settings are equivalent to the old ones, correct?

Comment (MilesBHuff) [reply tsconfig.json:14]: Not exactly.  ES6 is ES2015, but the original was 2017.  I called this out in the commit message:  the reason was that ES6 is more-widely supported than ES2017.  But we can change it back to ES2017 if you'd like.

As for `"lib"`:  Using `"ESNext"` means we don't have to list every single deviation from ES6 in the source code.  This was also called-out in the commit message.  ;)

File: tsconfig.json [hunk 4]
```diff
@@ -1,23 +1,72 @@
...
-        "lib": ["dom", "es2019", "es2020.string"],
+        // "baseUrl": ".",
+        // "outDir": "./dist/tsc",
+
+        // Library versions
+        "moduleResolution": "node",
         "module": "commonjs",
-        "noImplicitAny": false,
-        "target": "es2017",
+        "target": "ES6",
```
[the remark this answers is not in the document]
Comment (MilesBHuff) [reply tsconfig.json:11]: Ah!  Got it, I'll change that back.  Thanks for clarifying.

**Review by jjppof — CHANGES_REQUESTED** [review]
Comments added.

**Comment by MilesBHuff** [comment]
@jjppof Changes made.

**Review by jjppof — APPROVED** [review]

**Pull request merged**